util/feature/manager.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2021 PingCAP, Inc. Licensed under Apache-2.0.
+
+package feature
+
+import (
+	"fmt"
+	"net/http"
+	"sort"
+
+	"github.com/gin-gonic/gin"
+)
+
+type Manager struct {
+	version      string
+	flags        []*Flag
+	supportedMap map[string]struct{}
+}
+
+func NewManager(version string) *Manager {
```

Comment:
How about just calling this "VersionGuard"? Then it is normal to have a "GinFn" which does the version guard thing. We should avoid using "manager" as much as possible, if there is a more precise word to describe its behavior.